sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/urlExpressions.scala

```diff
@@ -86,16 +86,18 @@ case class UrlEncode(child: Expression)
   since = "3.4.0",
   group = "url_funcs")
 // scalastyle:on line.size.limit
-case class UrlDecode(child: Expression)
+case class UrlDecode(child: Expression, failOnError: Boolean = true)
   extends RuntimeReplaceable with UnaryLike[Expression] with ImplicitCastInputTypes {
 
+  def this(child: Expression) = this(child, true)
+
   override lazy val replacement: Expression =
     StaticInvoke(
       UrlCodec.getClass,
       SQLConf.get.defaultStringType,
       "decode",
-      Seq(child, Literal("UTF-8")),
-      Seq(StringTypeAnyCollation, StringTypeAnyCollation))
+      Seq(child, Literal("UTF-8"), Literal(failOnError)),
```

Comment:
Not related to this PR, but `Literal("UTF-8")` can be avoided here

Reply:
hmm how?

Reply:
move into the decode method

Reply:
oh, adding an overload of `decode` method that uses `utf8` as the default encoding? SGTM

Reply:
Thanks @yaooqinn , I will send a new PR to do it.

Reply:
This is the only place we call UrlCodec.decode and the `UTF-8` is the only charset we support